### PR TITLE
Amend/update frontend dependencies

### PR DESCRIPTION
**Pull request by M4R1KU**
This closes nothing

Summary of changes:
- ​updated nearly all frontend dependencies to the newest version
- cherry-picked factories from other branch

**important**
currently the build is only working correctly when the dependecies were installed with npm but not yarn
see [this](https://github.com/angular/angular-cli/issues/4594) issue

so you have to run yarn first and before you run `ng serve` or `ng build` you need to run `npm i`

I made sure to:

- [x] Add and / or update tests
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Update any documentation (javadoc, domain model, erd, ...)
- [x] Follow the contribution guidelines

Please review @jmesserli @needToRoll / cc: @outcobra/developers.

### CODE REVIEW

File: frontend/angular-cli.json [hunk 1]
```diff
@@ -1,64 +1,72 @@
 {
-    "project": {
-        "version": "1.0.0-beta.24",
-        "name": "frontend"
-    },
-    "apps": [
-        {
-            "root": "src",
-            "outDir": "dist",
-            "assets": [
```
Comment (jmesserli) [inline frontend/angular-cli.json]: do we actually use jquery or materialize?

**Comment by jmesserli** [comment]
can you please re-enable the ngBuild task in gradle?

**Comment by M4R1KU** [comment]
Why re-enable? It would fail tough expect we'd move to npm.

File: frontend/angular-cli.json (hunk 1 above)
Comment (M4R1KU) [inline frontend/angular-cli.json]: I'm not really sure. I need some time to check that but I don't have it.

**Comment by jmesserli** [comment]
I thought you fixed it, that's why

Mario Kunz <notifications@github.com> schrieb am Fr., 10. Feb. 2017, 22:33:

**Comment by M4R1KU** [comment]
What did I fix?
The factory: yes
But there is another error which is caused by yarn but don't ask me why this happens.

**Comment by jmesserli** [comment]
Okay it's fine then, you can merge it

Mario Kunz <notifications@github.com> schrieb am Fr., 10. Feb. 2017, 22:38:

**Comment by M4R1KU** [comment]
I hope that this error will be fixed in a future release of yarn or @angular/cli

**Comment by jmesserli** [comment]
We can hope ¯\_(ツ)_/¯

**Comment by jmesserli** [comment]
And I'm pretty sure that sentence does not exist in english

**Pull request merged**